akka-sample-distributed-data-scala/src/main/scala/sample/distributeddata/ReplicatedCache.scala

```diff
@@ -1,52 +1,65 @@
 package sample.distributeddata
 
-import akka.actor.Actor
-import akka.actor.ActorRef
-import akka.actor.Props
-import akka.cluster.Cluster
-import akka.cluster.ddata.DistributedData
+import akka.actor.typed.ActorRef
+import akka.actor.typed.scaladsl.Behaviors
 import akka.cluster.ddata.LWWMap
 import akka.cluster.ddata.LWWMapKey
+import akka.cluster.ddata.Replicator._
+import akka.cluster.ddata.SelfUniqueAddress
+import akka.cluster.ddata.typed.scaladsl.DistributedData
+import akka.cluster.ddata.typed.scaladsl.Replicator.{ Update, Get }
 
 object ReplicatedCache {
+  sealed trait Command
+  final case class PutInCache(key: String, value: Any) extends Command
+  final case class GetFromCache(key: String, replyTo: ActorRef[Cached]) extends Command
+  final case class Cached(key: String, value: Option[Any])
+  final case class Evict(key: String) extends Command
+  private sealed trait InternalCommand extends Command
+  private case class InternalGetResponse(key: String, replyTo: ActorRef[Cached], rsp: GetResponse[LWWMap[String, Any]])
+      extends InternalCommand
+  private case class InternalUpdateResponse(rsp: UpdateResponse[LWWMap[String, Any]]) extends InternalCommand
 
-  def props: Props = Props[ReplicatedCache]
+  def apply() = Behaviors.setup[Command] { context =>
```

Comment:
We try to use explicit type on the `apply(): Behavior[Command]`

and that probably means that you can remove the type annotation on `setup`